### PR TITLE
reduce sanitizer, handle for safe types, add additional tests

### PR DESCRIPTION
**Pull request by braxex**
## Description

@scwambach and I had attempted to reduce our API sanitization methods but ran into some tricky parts. This PR completes that work, adds handling for trusted safe types (number, boolean, undefined, null, NaN) and adds a few more tests 'cause who doesn't love more test coverage?

### How to test

1. Run the tests and celebrate their passing.
2. Confirm that calls to the database still pass nice data, and even strip out bad data.

### Changed Dependencies

n/a

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] ~~I have added analytics, if necessary~~
- [x] ~~I have updated the documentation, if necessary~~

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

**Comment by braxex** [comment]
Reviewer Note: Ignore failing CodeClimate diff. It's just mad because some code has been deleted.

File: services/app-api/utils/sanitize/sanitize.test.ts [hunk 1]
```diff
@@ -1,58 +1,141 @@
...
-      234234,
-      "<svg><g/onload=alert(2)//<p>",
-    ],
-    array2: [],
-  },
+
+// SAFE TYPES
+
+const safeBoolean = true;
+// const safeNaN = NaN;
```
Comment (karla-vm) [inline services/app-api/utils/sanitize/sanitize.test.ts:6]: Do we want to remove this commented line or add a `TODO` for later?

**Review by karla-vm — APPROVED** [review]
See comment about a commented line, otherwise lookin' good to me.

**Review by ntsummers1 — APPROVED** [review]

**Review by britt-mo — APPROVED** [review]

**Review by britt-mo — COMMENTED** [review]
Looks like there is a failing check

**Review by scwambach — APPROVED** [review]

File: services/app-api/utils/sanitize/sanitize.test.ts (hunk 1 above)
Comment (braxex) [reply services/app-api/utils/sanitize/sanitize.test.ts]: Added it back in. Must have accidentally commented it out. 🤦

**Pull request merged**